starter-code/src/components/CountryInfo.js

```diff
@@ -0,0 +1,74 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+
+class CountryInfo extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      country: this.props.location.state.listOfCountries
+    }
+  }
+
+  makeCountryNameArr = (countryCodeArr) => {
+    let countryNameArr = countryCodeArr.map(countryCode => this.getCountryData(countryCode).name.common)
+    return countryNameArr;
+  }
+
+  getCountryData = (countryCode) => {
+    let countryData = this.props.location.state.listOfCountries.find((country) => country.cca3 === countryCode);
+    return countryData;
+  }
+
+  render() {
+    let bordersArr = this.props.location.state.countryNameArr.map((borderCountry, index) => {
+      let countryData = this.getCountryData(this.props.location.state.borders[index]);
+      return <li key={index}>
+        <Link to={{
+          pathname: `/about/${this.props.location.state.borders[index]}`,
+          state: {
+            borders: countryData.borders,
+            countryName: countryData.name.common,
+            capital: countryData.capital,
+            area: countryData.area,
+            countryNameArr: this.makeCountryNameArr(countryData.borders),
+            listOfCountries: this.state.country
```

Comment:
read my comment on line 8 :) you can access the props here as well as you did in the state `this.props.location.state.listOfCountries`